test/orchard/inspect_test.clj

```diff
@@ -381,6 +381,37 @@
         (inspect/def-current-value *ns* "--test-val--"))
     (is (= 1 @(resolve '--test-val--)))))
 
+(deftest down-test
+  (testing "basic down"
+    (is (= :a (-> '{:foo [:a :b :c]}
+                  inspect
+                  (inspect/down 2)
+                  (inspect/down 1)
```

Comment:
Suggested extra coverage: try to go `down` out of boundaries, e.g. `inspect/down 99` in this line and see what happens

(we do that in `sibling*-test`)

Same for pagination-related tests

Reply:
Made the other 2 changes.
This one won't produce any predictable value as `down` fn doesn't have an out-of-boundary check. (It might be okay for now as a user cannot trigger this fn directly)

Reply:
`next-page` fn also has the same problem. Maybe this is what you meant with the first comment in this PR.

I will think about this a bit more and see if it's possible to add checks for both these fns.

Reply:
Thanks! Much appreciated

Reply:
Added out-of-bound checks for `down` and `next-page` fns and tried to cover all the scenarios in tests.

Reply:
Marking this as ready for review again. Feel free to suggest any more changes/tests/nits.